test/e2e-tests-examples.sh

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+
+# Copyright 2021 The Tekton Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script calls out to scripts in tektoncd/plumbing to setup a cluster
+# and deploy Tekton Pipelines to it for running integration tests.
+
+source $(dirname $0)/e2e-common.sh
+# Script entry point.
+
+set -u -e -o pipefail
+
+current_example=""
+yaml_files=""
+
+info() {
+  echo; echo "*** Example ${current_example}: $@ ***"; echo;
+}
+
+err() {
+  echo; echo "ERROR: Example ${current_example}: $@"; echo;
+}
+
+apply_files() {
+  info "Applying Resources"
+
+  folder=${REPO_ROOT_DIR}/examples/${current_example}
+  yaml_files=$(find ${folder}/ -name *.yaml | sort)
+  for y in ${yaml_files}; do
+    kubectl apply  -f ${y} || {
+      err "failed to apply ${y}"
+      exit 1
+    }
+  done
+}
+
+check_eventlistener() {
+  info "Waiting for EventListener to be available"
+
+  elName=${current_example}-listener
+  kubectl wait --for=condition=Available --timeout=20s eventlisteners/${elName} || {
+    err "eventlistener failed to get in running state"
+    exit 1
+  }
+  kubectl get eventlisteners
+}
+
+cleanup() {
+  info "Cleaning up resources"
+
+  for y in ${yaml_files}; do
+    kubectl delete -f ${y} || {
+      err "failed to delete ${y}"
+      exit 1
+    }
+  done
+}
+
+# Assumptions:
+# Name of example would be name of directory
+# Name of eventlistener must be (exampleName)-listener
+
+main() {
+  header "Starting Examples E2E Test"
+  folders="bitbucket github gitlab cron v1alpha1-task"
```

Comment:
I think we should be able to use most examples here. Two which probably won't work are the tls-connection one and the custom resource (knative) one.